language-design.md

```diff
@@ -28,7 +119,59 @@ you know a poorly designed language? What are the symptoms?
 
 **Response**
 
-
+*Well-designed*
+
+The most important characteristic of a well-designed language is that it should be able to 
+grow smoothly. In particular, this growth should be able to be carried out not just by the individuals 
+who initially designed the language but also by the many users who use it. This kind of growth is only 
+possible when there is a consistent pattern put in place by the designers, who “instead of designing a 
+thing” should “design a way of doing” [Steele, 1998]. When this kind of pattern exists, it becomes 
+easier for future users to identify and ultimately fill in the gaps, which allows the language to 
+expand and improve over time. 
+
+A well-designed language will never be able to completely satisfy every one of its users, but 
+it will allow for the flexibility for users to add the vocabulary and features that aren’t explicitly
+implemented in the language. Steele really emphasizes this importance of allowing users to define new words
+that both look and act like primitives in the language, and he mentions Lisp as a language that particularly 
+succeeds with this. The idea is to never make the user seem like they are powerless but, rather, to allow 
+them to feel that they exercise a degree of control over the language. 
+
+Another aspect of a well-designed language is one that balances being a readable and writable language. 
+For example, AppleScript is readable, but definitely not as writable. It is readable because it very closely 
```

Comment:
I think you're absolutely right! The assumption that the mimicry to the English language would be helpful for English-speaking programmers to write a program using language similar to the words they would say in English, but as you note, the costs of the flexibility outweighed the benefits of similarity.  The loss of precision makes it too difficult to write.

Also, in response to languages that are difficult to read and difficult to write, I like [Befunge](https://en.wikipedia.org/wiki/Befunge).  Wikipedia also keeps a list of various [Esoteric Programming Languages](https://en.wikipedia.org/wiki/Esoteric_programming_language).